File: java_projects/glue_code/commons-codec/src/main/java/org/apache/commons/codec/Charsets.java
package org.apache.commons.codec;

import java.nio.charset.Charset;
import java.nio.charset.StandardCharsets;
import org.graalvm.polyglot.Value;

public class Charsets {
  @Deprecated public static final Charset UTF_8 = StandardCharsets.UTF_8;
  @Deprecated public static final Charset UTF_16LE = StandardCharsets.UTF_16LE;
  @Deprecated public static final Charset UTF_16BE = StandardCharsets.UTF_16BE;
  @Deprecated public static final Charset UTF_16 = StandardCharsets.UTF_16;
  @Deprecated public static final Charset US_ASCII = StandardCharsets.US_ASCII;
  @Deprecated public static final Charset ISO_8859_1 = StandardCharsets.ISO_8859_1;
  private static Value clz = ContextInitializer.getPythonClass("/Charsets.py", "Charsets");
  private Value obj;

  public Charsets(Value obj) {
    this.obj = obj;
  }

  public Value getPythonObject() {
    return obj;
  }

  public static Charset toCharset1(final String charset) {
    //
    // return charset == null ? Charset.defaultCharset() : Charset.forName(charset);
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("toCharset1", charset).as(Charset.class);
  }

  public static Charset toCharset0(final Charset charset) {
    //
    // return charset == null ? Charset.defaultCharset() : charset;
    //

    // TODO: Check the type mapping below!
    return clz.invokeMember("toCharset0", charset).as(Charset.class);
  }
}

File: data/recomposed_projects/llama-3-70b-instruct/signature/commons-codec/src/main/org/apache/commons/codec/Charsets.py
from __future__ import annotations
import re
import io


class Charsets:

    UTF_8: str = "UTF_8"
    UTF_16: str = "UTF_16"
    US_ASCII: str = "US-ASCII"
    ISO_8859_1: str = "ISO-8859-1"
    UTF_16LE: str = "UTF_16LE"
    UTF_16BE: str = "UTF_16BE"

    @staticmethod
    def toCharset1(charset: str) -> str:
        return charset if charset is not None else Charset.defaultCharset()

    @staticmethod
    def toCharset0(charset: str) -> str:
        return charset if charset is not None else Charset.defaultCharset()
